src/main/java/io/trino/plugin/tiledb/TileDBSplitManager.java

```diff
@@ -238,13 +239,20 @@ private void GenerateCombinationTupleDomains(List<Pair<ColumnHandle, List<Domain
                 maxFromNonEmptyDomain = false;
             }
 
-            // If min is less than max, then this range cannot be split, as it returns zero records
+            // If min is greater than max, then this range cannot be split, as it returns zero records
             // This can happen if the query has a selection that puts an upper bound that is less than the
             // low bound, for example, SELECT * FROM ORDERS WHERE orderkey < 0
             if (min > max) {
                 return ranges;
             }
 
+            int j = 0;
+            while (((buckets - j) > 0) && ((max - min) / (buckets - j) == 0)) {
```

Comment:
@Shelnutt2 pointed out we shouldn't need a while loop here. IIUC, I think this can be written as `maximum((max-min % buckets), 1)`?

Reply:
Correct, and that is a lot nicer, but I still need to keep track of j to modify the number of buckets and use them in the following for loop in line 260. 
To explain a bit more: The number of buckets will only change if there are more buckets than the range needs. If not, which is the most common case, j == 0 and the algorithm works the same way as previously.